packages/query/src/index.ts

```diff
@@ -1168,23 +1168,26 @@ const generateQueryHook = async (
   let implementation = '';
   let mutators = undefined;
 
-  let isQuery =
-    Verbs.GET === verb &&
-    (override.query.useQuery ||
-      override.query.useSuspenseQuery ||
-      override.query.useInfinite ||
-      override.query.useSuspenseInfiniteQuery);
-
-  // Allows operationQueryOptions to override non-GET verbs
-  const hasOperationQueryOption =
-    operationQueryOptions?.useQuery ||
-    operationQueryOptions?.useSuspenseQuery ||
-    operationQueryOptions?.useInfinite ||
-    operationQueryOptions?.useSuspenseInfiniteQuery;
+  // Allows operationQueryOptions (which is the Orval config override for the operationId)
+  // to override non-GET verbs
+  const hasOperationQueryOption = !!(
+    operationQueryOptions &&
+    (operationQueryOptions?.useQuery ||
+      operationQueryOptions?.useSuspenseQuery ||
+      operationQueryOptions?.useInfinite ||
+      operationQueryOptions?.useSuspenseInfiniteQuery)
+  );
 
-  isQuery = isQuery || hasOperationQueryOption;
+  let isQuery =
+    (Verbs.GET === verb &&
+      (override.query.useQuery ||
+        override.query.useSuspenseQuery ||
+        override.query.useInfinite ||
+        override.query.useSuspenseInfiniteQuery)) ||
```

Comment:
ps im not even sure about this, the override.query.useQuery in all my cases even for gets and nothing/something in config overrides is always true :) So i thought it would just be `Verbs.GET === verb || hasOperationQueryOption` but im afraid to leave this part out since its introduced ...